lib/ret/api/credentials.ex

```diff
@@ -0,0 +1,115 @@
+defmodule Ret.Api.Credentials do
+  @moduledoc """
+  Credentials for API access.
+  """
+  alias Ret.Api.Credentials
+
+  alias Ret.Account
+
+  use Ecto.Schema
+  import Ecto.Query, only: [from: 2]
+  import Ecto.Changeset
+
+  @schema_prefix "ret0"
+  @primary_key {:api_credentials_id, :id, autogenerate: true}
+
+  schema "api_credentials" do
+    field(:api_credentials_sid, :string)
+    field(:token_hash, :string)
+    field(:subject_type, Ret.Api.TokenSubjectType)
+    field(:issued_at, :utc_datetime)
+    field(:expires_at, :utc_datetime)
+    field(:is_revoked, :boolean)
+    field(:scopes, {:array, Ret.Api.ScopeType})
+
+    belongs_to(:account, Account, references: :account_id)
+    timestamps()
+  end
+
+  def create_new_api_credentials(options) do
+    %Credentials{}
+    |> changeset(options)
+    |> Ret.Repo.insert()
+  end
+
+  defp changeset(%Credentials{} = credentials, %{
+         token_hash: token_hash,
+         subject_type: subject_type,
+         scopes: scopes,
+         account: account
+       }) do
+    credentials
+    |> change()
+    |> add_api_credentials_sid_to_changeset
+    |> add_token_hash_to_changeset(token_hash)
+    |> add_subject_type_to_changeset(subject_type)
+    |> add_issued_at_to_changeset(Timex.now() |> DateTime.truncate(:second))
+    |> add_expires_at_to_changeset(Timex.now() |> Timex.shift(years: 1) |> DateTime.truncate(:second))
```

Comment:
I don't know what an appropriate expiration time would be. I've read best practice advice that suggests that we should access tokens that expire quickly and refresh tokens that are used once (then revoked shortly thereafter) to generate a new `{access_token, refresh_token}` pair. However, I have not implemented `refresh_token`s (and we may want to change the implementation of `access_token`s a bit to make adding `refresh_token`s easier).

Reply:
I think access and refresh tokens make sense when you're doing an oauth flow, but I'm not sure they make sense in the context of a conventional API, unless this is a GraphQL convention that I'm not aware of. For example, it would be quite strange to me if my Github tokens expired after some arbitrary period. 
I'd lean towards removing token expiry altogether and relying on users to revoke their tokens manually when needed.

Reply:
Ok, removed expiration